### PR TITLE
Patch for microphone capture

### PR DESCRIPTION
**Pull request by yunlong-wang-cn**
link to this issue #175,

Sometimes the resource of the microphone is occupied by some unclear reason, at that time you can't specify the hardware for audio_capture, but you wait until the resource release.

### CODE REVIEW

File: marimbabot_audio/launch/marimbabot.launch [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
     <node name="audio_from_lilypond" pkg="marimbabot_audio" type="audio_from_lilypond.py"/>
-    <node name="audio_audio_capture" pkg="audio_capture" type="audio_capture" output="screen">
-        <param name="device" value="" />
-        <param name="format" value="wave" />
-        <param name="channels" value="1" />
-        <param name="depth" value="16" />
-        <param name="sample_rate" value="44100" />
-    </node>
+=======
+>>>>>>> edd50390e426be44db374a26a50dbddac9826170
```
Comment (Flova) [inline marimbabot_audio/launch/marimbabot.launch:8]: It seems like you committed a merge conflict

**Review by yunlong-wang-cn — COMMENTED** [review]
the last commit actually merges the updates from the main, and it cause the conflict

File: marimbabot_audio/scripts/onset_detection.py [hunk 2]
```diff
@@ -438,7 +443,8 @@ def update_spectrogram(self, spec,onsets_cqt, ys, xs, durations_len):
         if self.pub_spectrogram.get_num_connections() == 0:
             self.spectrogram = None
             return
-        
+
+        # rospy.logdebug("update spectrogram")
```
Comment (Flova) [inline marimbabot_audio/scripts/onset_detection.py:447]: Remove commented out code

File: marimbabot_audio/scripts/onset_detection.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
+# LD_LIBRARY_PATH=/home/wang/workspace/marimbabot_ws/env_this/lib/python3.8/site-packages/nvidia/cudnn/lib:$LD_LIBRARY_PATH
```
Comment (Flova) [inline marimbabot_audio/scripts/onset_detection.py:1]: Why is this in here?

File: marimbabot_audio/scripts/onset_detection.py [hunk 4]
```diff
@@ -520,6 +526,6 @@ def publish_cqt(self, cqt):
         self.pub_cqt.publish(msg)
 
 if __name__ == '__main__':
-    rospy.init_node("detect_onset")
+    rospy.init_node("detect_onset",log_level=rospy.DEBUG)
```
Comment (Flova) [inline marimbabot_audio/scripts/onset_detection.py:529]: Why is the loglevel hardcoded to debug?

File: marimbabot_audio/launch/marimbabot.launch (hunk 1 above)
Comment (yunlong-wang-cn) [reply marimbabot_audio/launch/marimbabot.launch:8]: working on it

File: marimbabot_bringup/launch/marimbabot.launch [hunk 5]
```diff
@@ -1,7 +1,14 @@
 <launch>
-  <include file="$(find marimbabot_vision)/launch/marimbabot.launch"/>
+  <!-- <include file="$(find marimbabot_vision)/launch/marimbabot.launch"/> -->
   <include file="$(find marimbabot_audio)/launch/marimbabot.launch"/>
+<<<<<<< HEAD
```
Comment (Flova) [inline marimbabot_bringup/launch/marimbabot.launch:4]: More unmerged files

File: marimbabot_speech/src/speech2text.py [hunk 6]
```diff
@@ -1,4 +1,6 @@
 import os
+from nvidia import cudnn
+os.environ['LD_LIBRARY_PATH'] = f'{os.path.dirname(cudnn.__file__)}/lib:$LD_LIBRARY_PATH'
```
Comment (Flova) [inline marimbabot_speech/src/speech2text.py:3]: Can you add a comment why this change is needed?

**Review by Flova — CHANGES_REQUESTED** [review]

**Comment by Flova** [comment]
> the last commit actually merges the updates from the main, and it cause the conflict

Okay :+1:

**Review by yunlong-wang-cn — COMMENTED** [review]
solve conflict

File: marimbabot_speech/src/speech2text.py (hunk 6 above)
Comment (yunlong-wang-cn) [reply marimbabot_speech/src/speech2text.py:3]: since I haven't specified the version of TensorFlow, it seems to install the newest version, so according to [here](https://www.tensorflow.org/install/pip), the environment needs to be configured.

File: marimbabot_audio/launch/marimbabot.launch [hunk 7]
```diff
@@ -1,14 +1,19 @@
...
+          <param name="channels" value="1" />
+          <param name="depth" value="16" />
+          <param name="sample_rate" value="44100" />
+      </node>
+    </group>
     <node name="onset_detector" pkg="marimbabot_audio" type="onset_detection.py" output="screen" />
     <node name="onset_midi" pkg="marimbabot_audio" type="note2midi.py" output="screen" />
     <node name="soundplay_node" pkg="sound_play" type="soundplay_node.py"/>
+    <node name="audio_from_lilypond" pkg="marimbabot_audio" type="audio_from_lilypond.py"/>
+    <node name="audio_node_rviz" pkg="rviz" type="rviz" args="-d $(find marimbabot_audio)/config/config.rviz"/>
```
Comment (Flova) [inline marimbabot_audio/launch/marimbabot.launch:18]: Do we want to launch rviz every time?

File: marimbabot_audio/launch/marimbabot.launch [hunk 8]
```diff
@@ -1,14 +1,19 @@
...
-        <param name="format" value="wave" />
-        <param name="channels" value="1" />
-        <param name="depth" value="16" />
-        <param name="sample_rate" value="44100" />
-    </node>
+
+  <!-- list the devices: arecord -l -->
+    <group ns="audio_node">
+      <node name="note_audio_capture" pkg="audio_capture" type="audio_capture" output="screen">
+          <param name="device" value="plughw:3,0" />
```
Comment (Flova) [inline marimbabot_audio/launch/marimbabot.launch:7]: This seems very device specific

File: marimbabot_audio/scripts/onset_detection.py (hunk 4 above)
Comment (yunlong-wang-cn) [reply marimbabot_audio/scripts/onset_detection.py:529]: If you want to turn on the dedug info, you can choose not to output to screen in launch file, right? or what's your opionin?

File: marimbabot_bringup/launch/marimbabot.launch [hunk 9]
```diff
@@ -1,5 +1,5 @@
 <launch>
-  <include file="$(find marimbabot_vision)/launch/marimbabot.launch"/>
+  <!-- <include file="$(find marimbabot_vision)/launch/marimbabot.launch"/> -->
```
Comment (Flova) [inline marimbabot_bringup/launch/marimbabot.launch:2]: Why did you remove the vision?

File: marimbabot_audio/launch/marimbabot.launch (hunk 7 above)
Comment (yunlong-wang-cn) [reply marimbabot_audio/launch/marimbabot.launch:18]: ok, let me create a new launch file for exhibition, like the vision node.

File: marimbabot_speech/src/speech2text.py (hunk 6 above)
Comment (Flova) [reply marimbabot_speech/src/speech2text.py:3]: Okay :+1: please add this as a comment next to the loc

File: marimbabot_audio/launch/marimbabot.launch (hunk 8 above)
Comment (yunlong-wang-cn) [reply marimbabot_audio/launch/marimbabot.launch]: yes, it seems the only way to specify the hardware currently, and the sequence to plugin the used device could change the id of the device, it is indeed uncomfortable.

File: marimbabot_bringup/launch/marimbabot.launch (hunk 9 above)
Comment (yunlong-wang-cn) [reply marimbabot_bringup/launch/marimbabot.launch:2]: that is a accident, sorry.

File: marimbabot_audio/scripts/onset_detection.py (hunk 4 above)
Comment (Flova) [reply marimbabot_audio/scripts/onset_detection.py:529]: Exactly. ROS has multiple ways of setting the log level, but I would not hardcode it.

File: marimbabot_speech/src/speech2text.py (hunk 6 above)
Comment (yunlong-wang-cn) [reply marimbabot_speech/src/speech2text.py:3]: You mean to add the comment next to this code, right?

**Review by Flova — APPROVED** [review]
